processor/transformprocessor/internal/metrics/func_copy_metric.go

```diff
@@ -0,0 +1,67 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package metrics // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/metrics"
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlmetric"
+)
+
+type copyMetricArguments struct {
+	Name        ottl.Optional[ottl.StringGetter[ottlmetric.TransformContext]]
+	Description ottl.Optional[ottl.StringGetter[ottlmetric.TransformContext]]
+	Unit        ottl.Optional[ottl.StringGetter[ottlmetric.TransformContext]]
+}
+
+func newCopyMetricFactory() ottl.Factory[ottlmetric.TransformContext] {
+	return ottl.NewFactory("copy_metric", &copyMetricArguments{}, createCopyMetricFunction)
+}
+
+func createCopyMetricFunction(_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[ottlmetric.TransformContext], error) {
+	args, ok := oArgs.(*copyMetricArguments)
+
+	if !ok {
+		return nil, fmt.Errorf("extractCountMetricFactory args must be of type *extractCountMetricArguments")
```

Comment:
```suggestion
		return nil, fmt.Errorf("createCopyMetricFunction args must be of type *copyMetricArguments")
```
Aside: sometimes I wonder whether we should just remove these errors, panic instead, and enforce testing that makes sure we are virtually guaranteed not to hit this case. If the parser can instantiate this function in a test, it will also work at runtime.

Reply:
Ya good idea, this has turned out to be unhelpful boilerplate